patches/minecraft/net/minecraft/client/particle/TerrainParticle.java.patch

```diff
@@ -1,17 +1,29 @@
 --- a/net/minecraft/client/particle/TerrainParticle.java
 +++ b/net/minecraft/client/particle/TerrainParticle.java
-@@ -69,7 +_,13 @@
+@@ -16,6 +_,7 @@
+    private final float f_108277_;
+    private final float f_108278_;
+ 
++   @Deprecated // Forge: Use BlockPos aware constructor below
+    public TerrainParticle(ClientLevel p_108282_, double p_108283_, double p_108284_, double p_108285_, double p_108286_, double p_108287_, double p_108288_, BlockState p_108289_) {
+       this(p_108282_, p_108283_, p_108284_, p_108285_, p_108286_, p_108287_, p_108288_, p_108289_, new BlockPos(p_108283_, p_108284_, p_108285_));
+    }
+@@ -23,7 +_,8 @@
+    public TerrainParticle(ClientLevel p_172451_, double p_172452_, double p_172453_, double p_172454_, double p_172455_, double p_172456_, double p_172457_, BlockState p_172458_, BlockPos p_172459_) {
+       super(p_172451_, p_172452_, p_172453_, p_172454_, p_172455_, p_172456_, p_172457_);
+       this.f_108280_ = p_172459_;
+-      this.m_108337_(Minecraft.m_91087_().m_91289_().m_110907_().m_110882_(p_172458_));
++      if (p_172459_ == null) this.m_108337_(Minecraft.m_91087_().m_91289_().m_110907_().m_110882_(p_172458_));
++      else this.m_108337_(Minecraft.m_91087_().m_91289_().m_110907_().getTexture(p_172458_, p_172451_, p_172459_));
```

Comment:
We can't do this. As you can see above, the BlockPos is instantiated with the x y z pos if it is not passed in, which is not always the position of the block, but the position where the particles should render. Block pos may and will be != particle pos in some cases and ause wrong lookups, which we definetly don't want. The other commit you mentioned looks better, as it passes the correct BlockPos.

Reply:
Yeah thanks, you're right, realised this after. I've switched back to my first solution which works and is correct the only issue being if it was ok for `updateSprite` to be public facing